pkg/collector/corechecks/cluster/kubernetes_state.go

```diff
@@ -211,7 +228,45 @@ func (c *KSMConfig) parse(data []byte) error {
 }
 
 // Run runs the KSM check
+// This is a long-running check, the Run method is blocking and handles the scheduling of metrics collection
 func (k *KSMCheck) Run() error {
+	runTicker := time.NewTicker(k.collectionInterval)
+	defer runTicker.Stop()
+
+	// Do not wait for the first tick and start the first metrics collection ASAP
+	k.collectWithStats()
+
+	for {
+		select {
+		case <-runTicker.C:
+			log.Tracef("Collecting KSM metrics: '%s'", k.ID())
+			k.collectWithStats()
+		case <-k.ctx.Done():
+			log.Debugf("Stopping KSM check '%s'", k.ID())
+			return nil
+		}
+	}
+}
+
+// collectWithStats executes one KSM check run and updates its stats
+func (k *KSMCheck) collectWithStats() {
+	t0 := time.Now()
+
+	checkErr := k.collect()
+	if checkErr != nil {
+		log.Debugf("Error running check '%s': %v", k.ID(), checkErr)
+	}
+
+	stats, statsErr := k.GetMetricStats()
+	if statsErr != nil {
+		log.Debugf("Couldn't collect stats from check '%s': %v", k.ID(), statsErr)
+	}
+
+	runner.AddWorkStats(k, time.Since(t0), checkErr, []error{}, stats)
```

Comment:
maybe you should create `t1 := time.Now()` before  [line 260](https://github.com/DataDog/datadog-agent/pull/6540/files#diff-9639de9e2bd5f43e4d1fe996b6763ae5R260),  to not take into consideration the stats aggregation duration.

and then
```suggestion
	runner.AddWorkStats(k, t1.Sub(t0), checkErr, []error{}, stats)
```

Reply:
Based on the regular check stats [here](https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/runner/runner.go#L268-L317), the time collecting stats is also considering as check time, I want to keep things consistent here